### PR TITLE
Nehal internship project blog

### PR DESCRIPTION
**Pull request by N-Shar-ma**

### CODE REVIEW

File: blog/2021-12-15-new-marketplace/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Figuring out the Root of the Problem:
+
+Taking the most obvious first step, I googled the error, and as expected, multiple ways showed up which worked differently for different projects. And no amount of googling could explain the reason this error occurred only sometimes. I then turned to experimentation, and spent a lot of time importing the apps package in different environments, while also reading about how bundling works and the intricacies of imports and exports. I tried importing apps individually too, where I saw *another* new error pop up! Though it was admittedly frustrating to spend days documenting the cases imports worked and the cases they didn't, without any measurable outcome, this method did pay off and granted me a moment of near enlightenment when I observed a pattern and realized that the apps package import only failed in node / server side environments! The new error too, I noticed, only cropped up when the app imported had any asynchronous functionality.
```
Comment (hargup) [inline blog/2021-12-15-new-marketplace/index.md:18]: Let's replace "googled", with "searched"

File: blog/2021-12-15-new-marketplace/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+title: Creating a Self Updating Marketplace to Showcase all Felvin Instant Apps
```
Comment (hargup) [inline blog/2021-12-15-new-marketplace/index.md:2]: Let's make the title catchy, suggestions:

- "How many instant apps do we have at Felvin? There is an app for that!"
- "An App to show all the apps"
- "How did I create an app to show all the available apps?"

File: blog/2021-12-15-new-marketplace/index.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## A Surprise Bonus
+
+Until I made the Marketplace self updating, I thought of extracting the data of all current Instant Apps to have a somewhat updated Marketplace in its lieu. While trying out ways to do so, I ended up making a temporary (or so I initially thought) Instant App to provide me the current data of all the apps in JSON format. Soon I realized that with some UI changes, this Instant App can be used as an alternative showcase for all Instant Apps! Anytime you ~~search~~ *felvin* something like "[Felvin Instant Apps](https://felvin.com/search?q=Felvin%20Instant%20Apps)" or "[what instant apps are available](https://felvin.com/search?q=what%20instant%20apps%20are%20available)" a carousel showcasing all the Instant Apps will appear. 
+
+An Instant App of Instant Apps... isn't that so meta?!
```
Comment (hargup) [inline blog/2021-12-15-new-marketplace/index.md:26]: Love the end 😄

**Review by hargup — APPROVED** [review]

**Comment by hargup** [comment]
@N-Shar-ma a screenshot of the instant app marketplace will be great!

**Review by sahil-shubham — APPROVED** [review]

**Pull request merged**